docs/design/fsdp-design-en.md

```diff
@@ -0,0 +1,373 @@
+# DLRover's FSDP Elastic Training Design Document
+
+## 1. Background
+
+Large-scale machine learning models have expanded, necessitating
+distributed training. DLRover excels in scenarios like node failure,
+dynamic resource allocation, and automatic recovery. While PyTorch
+supports FSDP (Fully Sharded Data Parallel), DLRover faces challenges
+with dynamic resources and fault tolerance. Issues include OOM (Out of
+Memory) and difficulties reloading sharded checkpoints when world size
```

Comment:
The sentence has a grammar error.

Reply:
Thanks for catching those grammatical issues in the design doc. I've addressed and updated them. Please review the latest commit.